[DON'T MERGE] Category style for ReviewApp

okkez opened the pull request:
For https://github.com/bm-sms/daimon-news/pull/582

okkez committed:
Select multiple categories

okkez committed:
Follow previous changes related to Groonga

okkez committed:
Follow previous changes related to view

okkez committed:
Update seed to use multiple categories

okkez committed:
Fix validation

okkez committed:
Setup order value according to target

okkez committed:
Add scope to fetch categorized posts

okkez committed:
Use categories

okkez committed:
Use categorized scope

But this action does not work well...

okkez committed:
Use categories

okkez committed:
Use `.search-result-list`

Because we have multiple categories under another `<ul>`.

okkez committed:
Fix integration test for multiple categories

okkez committed:
Simplify CurrentCategoryTest

We don't need register records via management console.

okkez committed:
Add parentheses

okkez committed:
Add parentheses

okkez committed:
Use categories

okkez committed:
Use `<ul>` to display categories on `/editor/posts`

okkez committed:
Set `list-style-type: none` in `/editor/posts`

okkez committed:
Set `categorization.order` properly

okkez committed:
Apply toArray for jQuery object

In previous version, `Array.from(orders)` does not work on phantomjs.
Because phantomjs does not support ES2015(ES6) features.
So I convert jQuery object to Array before apply _toConsumableArray.

okkez committed:
Set categories via categorizations

okkez committed:
Remove needless code

Follow previous changes.

okkez committed:
Display related categories after content

okkez committed:
Fix scope Post.categorized

In this version, we can fetch categorized posts that have multiple
categories.

okkez committed:
Add seed

Posts have multiple categories.

okkez committed:
Remove category description from category list

Because it is redundant.

okkez committed:
Fix a typo

okkez committed:
Arrange design

okkez committed:
Display recent posts in `categories#show`

okkez committed:
Display related categories before related posts

okkez committed:
Extract partial template for related categories

okkez committed:
Use ordered list (ol)

okkez committed:
Do not display categories that have no published posts

Display category when the category's children have published posts.

okkez committed:
Display root categories properly

In previous version, display category that has no posts.

okkez committed:
Fix `Category#post_count`

Count posts in all subtree.

okkez committed:
Display post that does not have categories properly

tricknotes commented on app/views/categories/show.html.haml:
```diff
@@ -22,5 +22,6 @@
-  - else
-    = render 'posts', posts: @posts
+
+  %h2 このカテゴリの新着記事
```
Where does this code come from?
I couldn't find in https://github.com/bm-sms/daimon-news/pull/582

okkez replied:
in my category-design branch.

---

今、ローカルにあるブランチです。デザイン関係は別PRにしようとしてました。

tricknotes replied:
Please push it to remote repository :-)

tricknotes replied:
However this message is unnecessary, I think.

okkez committed:
Use pre-loaded record object

okkez committed:
Simplify

okkez committed:
Remove unnecessary heading

okkez committed:
Cosmetic changes

okkez committed:
Rename scope

categorized ->
categorized_by

okkez committed:
Use `lambda` instead of `->`

This suppresses rubocop warnings such as:

* Style/Lambda: Use the lambda method for multiline lambdas.
* Style/IndentationWidth: Use 2 (not -13) spaces for indentation.

okkez committed:
Extract migration for #582

We should merge this commit before merge #582

okkez committed:
Add models related to multiple categories

okkez committed:
Remove null constraint from `posts.category_id`

Because we will change seed data in #582.

okkez committed:
Remove microdata from related categories

okkez committed:
Use `pluck(:id)` instead of `select(:id)`

Because `select(:id)` returns AR object.

okkez committed:
Assume that `category_ids` is an `Array`

okkez committed:
Use String instead of Symbol

okkez committed:
Highlight root category of current category on menu bar

okkez committed:
Use `#each` instead of `#map`

tricknotes committed:
Add validation for `Post#categorizations` to have one or more item

tricknotes committed:
Update seed to create post with category in atomically

okkez committed:
Set current category properly

okkez committed:
Add `CategoryDecorator#highlighted?`

We cannot assume that current_category is present.
Because `CurrentResourceHelper#current_category` returns `nil`.

tricknotes committed:
Fix test to create categorization with post

tricknotes committed:
Show error message for categorization

okkez committed:
Fix custom validation

In previous version, we can register post with category has children.

okkez committed:
Add validation

In previous version, raise error when create/update post that has same
category twice.

okkez committed:
Use meaningful name

okkez committed:
Use Post custom validation instead of Categorization's validation

Because in previous version, we cannot validate multiple same categories
when create/update Post.

okkez committed:
Category is required now

See #599

okkez committed:
Add `Post#main_category` and use it

okkez committed:
Remove useless assignment

okkez committed:
Suppress rubocop warning Style/MultilineMethodCallBraceLayout

okkez committed:
Use feature flag for multiple categories

tricknotes committed:
Fix markup to keep backward compatibility

tricknotes committed:
Add sub header for posts

tricknotes committed:
[Refactor] Move header to partial

tricknotes committed:
[Refactor] `next` in view is difficult to me

tricknotes committed:
[Refactor] Make partial inline

Partial should be a private view in controller scope.

tricknotes committed:
[Refactor] Use class name to apply style

okkez committed:
Stop displaying duplicated posts on `categories#show` post list when it has multiple categories

In previous version, duplicated posts are displayed.

tricknotes committed:
Specify class name to independ content type

This list may have post or category.
However style is shared these items.

okkez committed:
Add feature flag for hierarchical categories

tricknotes committed:
Fix sub category's text :<

okkez committed:
Revert "Use `to_sql` to reduce SQL"

This reverts commit 6546a6aadf8fb3cbe05da323cb436849c7906d82.

Because it does not work.

Generated SQL as following:

```
SELECT
 "categories"."id" AS t0_r0,
 "categories"."name" AS t0_r1,
 "categories"."description" AS t0_r2,
 "categories"."created_at" AS t0_r3,
 "categories"."updated_at" AS t0_r4,
 "categories"."site_id" AS t0_r5,
 "categories"."slug" AS t0_r6,
 "categories"."order" AS t0_r7,
 "categories"."ancestry" AS t0_r8,
 "posts"."id" AS t1_r0,
 "posts"."title" AS t1_r1,
 "posts"."body" AS t1_r2,
 "posts"."site_id" AS t1_r3,
 "posts"."created_at" AS t1_r4,
 "posts"."updated_at" AS t1_r5,
 "posts"."published_at" AS t1_r6,
 "posts"."category_id" AS t1_r7,
 "posts"."thumbnail" AS t1_r8,
 "posts"."original_updated_at" AS t1_r9,
 "posts"."public_id" AS t1_r10,
 "posts"."serial_id" AS t1_r11
FROM
 "categories"
 LEFT OUTER JOIN
 "categorizations"
   ON "categorizations"."category_id" = "categories"."id"
 LEFT OUTER JOIN
 "posts"
   ON "posts"."id" = "categorizations"."post_id"
WHERE
 "posts"."id" IS NULL
 AND ("categories"."id" != $1)
 ORDER BY "categories".ancestry NULLS FIRST, "categories"."order" ASC  [["id", 0]]
```

tricknotes committed:
Add more link

okkez committed:
Use feature flag hierarchical_categories_enabled

tricknotes committed:
Revert "Use feature flag for multiple categories"

This reverts commit b9444fcb781d3cde5faeee4f52e9ca6597c676b0.
This feature doesn't break anything.

tricknotes committed:
Specify `Site#hierarchical_categories_enabled` flag to development seed

tricknotes committed:
Add class to apply style

tricknotes committed:
[Refactor] Remove dup of DOM structure

tricknotes committed:
Fix current link in related categories' breadcrumbs

tricknotes committed:
[Refactor] Don't pass extra option to gretel

tricknotes committed:
Update microdata using http://schema.org

tricknotes committed:
Add missing tailing semicolon.

tricknotes committed:
[Refactor] :golf:

tricknotes committed:
[Refactor] `uniq` should be called with `joins` in this case

tricknotes committed:
Update comment

We should remove this method when a new ancestry is released.

mtsmfm committed:
Cosme DOM structure

tricknotes committed:
[Refactor] Don't use view specific method name

mtsmfm committed:
Use category-related-design branch

tricknotes committed:
[Refactor] Remove unused local variables

mtsmfm committed:
Merge pull request #607 from bm-sms/update-microdata

Update microdata

tricknotes commented:
Thanks Review Apps ❤️ 
This feature is landed on master 🚀
